packages/components/src/internal/query/selectRows.ts

```diff
@@ -79,3 +79,17 @@ export async function selectRows(options: SelectRowsOptions): Promise<SelectRows
         schemaQuery,
     };
 }
+
+/**
+ * selections: The selections object from a QueryModel
+ */
+interface GetSelectedRowsOptions extends SelectRowsOptions {
+    keyColumn?: string;
+    selections: Set<string>;
+}
+
+export function getSelectedRows(options: GetSelectedRowsOptions): Promise<SelectRowsResponse> {
+    const { keyColumn = 'RowId', selections, ...rest } = options;
+    const filterArray = [Filter.create(keyColumn, Array.from(selections), Filter.Types.IN)];
```

Comment:
Either `Omit` the `filterArray` option or allow for the passed in `filterArray` to be appended to.